lib/services/service-utilities.ts

```diff
@@ -0,0 +1,204 @@
+	private _eventListeners: { [event: string]: ServiceRegistration[] } = {};
+
+	/**
+	 * Store an event of interest, so that the method gets triggered appropriately.
+	 * @param registration  Registration object with event trigger and other details.
+	 */
+	public registerEvent(registration: ServiceRegistration): void {
+		// Store each event registration in an object of arrays.
+		for (const event of registration.events) {
+			if (!this._eventListeners[event]) {
+				this._eventListeners[event] = [];
+			}
+			this._eventListeners[event].push(registration);
+		}
+	}
+
+	/**
+	 * Emit data to the service.
+	 * @param data  Service Emit Request to send, if relevant.
+	 * @returns     Details of the successful transmission from the service.
+	 */
+	public sendData(data: ServiceEmitRequest): Promise<ServiceEmitResponse> {
+		const context = data.contexts[this.serviceName] as ServiceEmitContext;
+		if (!context) {
+			return Promise.resolve({
+				err: new ContextAbsentError(`No ${this.serviceName} context`),
+				source: this.serviceName,
+			});
+		}
+
+		return this.emitData(context).then((response: any) => {
+			return {
+				response,
+				source: this.serviceName,
+			};
+		}).catch((err: TypedError) => {
+			return {
+				err,
+				source: this.serviceName,
+			};
+		});
+	}
+
+	/**
+	 * Queue an event ready for running in a child, here to type guard.
+	 * @param data  The WorkerEvent to add to the queue for processing.
+	 */
+	protected queueData = (data: UtilityServiceEvent) => {
+		if (this.verify(data)) {
+			super.queueEvent({
+				data,
+				workerMethod: this.handleEvent
+			});
+		} else {
+			this.logger.log(LogLevel.WARN, `Event failed verification.`);
+		}
+	}
+
+	/**
+	 * Emit a payload to the endpoint defined, resolving when done.
+	 * endpoint  Definition of the endpoint to emit to.
+	 * payload   Data to be delivered.
+	 */
+	protected abstract emitData(data: ServiceEmitContext): Promise<any>;
+
+	/**
+	 * Verify the event before enqueueing.  A naive approach could be to simply return true, but that must be explicit.
+	 * @param data  The data object that was enqueued.
+	 */
+	protected abstract verify(data: UtilityServiceEvent): boolean
+
+	/**
+	 * Get a Worker object for the provided event, threaded by context.
+	 * @param event  Event as enqueued by the listener.
+	 * @returns      Worker for the context associated.
+	 */
+	protected getWorker = (event: UtilityWorkerEvent): Worker<T> => {
+		// Attempt to retrieve an active worker for the context
+		const context = event.data.context;
+		const retrieved = this.workers.get(context);
+		if (retrieved) {
+			return retrieved;
+		}
+		// Create and store a worker for the context
+		const created = new Worker<T>(context, this.removeWorker);
+		this.workers.set(context, created);
+		return created;
+	}
+
+	/**
+	 * Create or retrieve the express app.
+	 * @returns  Express server app.
+	 */
+	protected get expressApp(): express.Express {
```

Comment:
I think there needs to be the ability to pass an already existing Express instance in here, similar in style to what the `front-sdk` package does.